### ISSUE TITLE
meta-data dictionary lost when using bracket operator

### ISSUE THREAD

**Issue by zivy** [SimpleITK/SimpleITK#1911]
This is a bug with the bracket operator having a side effect, loss of the meta-data dictionary.

Sample code:
```
import SimpleITK as sitk

image = sitk.Image([2,3], sitk.sitkUInt8)
image.SetMetaData("a","1")
image[0,0] = 8
print(f"after setting single value: expected True got {image.HasMetaDataKey('a')}") # metadata exists

image[0:1,1:3] = 8
print(f"after setting region value: expected True got {image.HasMetaDataKey('a')}") # metadata lost

image.SetMetaData("a","1")
mask = (image == 8)
image[mask] = 9
print(f"after setting region value: expected True got {image.HasMetaDataKey('a')}") # metadata lost

```

**Comment by zivy** [comment]
Minor update to the prints so that it reports what is expected (always True) and what actually happens.

**Comment by blowekamp** [comment]
fixed by #1920
